service/internal/pipelines/pipelines.go

```diff
@@ -0,0 +1,169 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pipelines // import "go.opentelemetry.io/collector/service/internal/pipelines"
+
+import (
+	"context"
+	"net/http"
+	"sort"
+
+	"go.uber.org/multierr"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/service/internal/builder"
+	"go.opentelemetry.io/collector/service/internal/zpages"
+)
+
+const (
+	zPipelineName  = "zpipelinename"
+	zComponentName = "zcomponentname"
+	zComponentKind = "zcomponentkind"
+)
+
+// BuiltPipelines is set of all pipelines created from exporter configs.
+type BuiltPipelines struct {
+	telemetry component.TelemetrySettings
+
+	exporters *builder.BuiltExporters
+	pipelines builder.BuiltPipelines
+	receivers builder.Receivers
+}
+
+// Build builds all pipelines from config.
+func Build(ctx context.Context, telemetry component.TelemetrySettings, buildInfo component.BuildInfo, cfg *config.Config, factories component.Factories) (*BuiltPipelines, error) {
```

Comment:
It feels weird to pass both `component.TelemetrySettings` and `service.Config` as arguments, given that the config struct has `service.ConfigServiceTelemetry` within it. I definitely want to be able to customize the `component.TelemetrySettings` for my use case, and I think several other people have expressed their need for this, so not passing the telemetry structs in some way does not feel like an option. Some possibilities are:

1. Move `service.ConfigServiceTelemetry` out of its current location in the configuration. I think this is not a realistic option if it affects the YAML, since it will break too many things, but we could do it with a custom unmarshaling function if we want to go this way.
2. Do a 'telemetry provider' kind of thing. It makes this somewhat more clunky to use for e.g. my use case, but maybe it's the way to go?
3. Do nothing and just document it; accept that we can't change it and that we have this weirdness in the API

Reply:
Will address this issue separately, I am planning to actually send only what is needed (no telemetry settings).

Reply:
👍 , yeah, thinking about this later, the `Extensions` field is also a problem, so probably we can just send exactly what we need